src/Workspaces/Remote/Core/ServiceDescriptor.cs

```diff
@@ -78,6 +79,26 @@ protected override JsonRpcConnection CreateConnection(JsonRpc jsonRpc)
             return connection;
         }
 
+        public override ServiceRpcDescriptor WithMultiplexingStream(MultiplexingStream? multiplexingStream)
+        {
+            var baseResult = base.WithMultiplexingStream(multiplexingStream);
+            if (baseResult is ServiceDescriptor)
+                return baseResult;
```

Comment:
📝 This path is always used by 16.8 Preview 3 and newer. The rest of this method is an inlined version of the fixed code. I'll submit a follow-up pull request to remove this method as soon as this pull request merges, and we can merge that follow-up as soon as roslyn-integration-CI is updated to 16.8 Preview 3 or newer.